pkg/gardenlet/controller/federatedseed/seed_control.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package federatedseed
+
+import (
+	"context"
+	"sync"
+
+	"github.com/gardener/gardener/pkg/apis/core/v1alpha1/helper"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
```

Comment:
That's tricky because currently ShootState is in alpha and in some files we have to import both core/v1alpha1 and core/v1beta1. Maybe we have to stick to one api version, I am not sure?

Reply:
Hmm, okay, the `ShootState` resource we have to read/write with `core/v1alpha1`, of course, sure, but all the other things should be done with `v1beta1` already (maybe it's already the case, I have not checked).